modules/dnn/src/int8layers/quantization_utils.cpp

```diff
@@ -35,43 +38,123 @@ static void broadcast1D2TargetMat(Mat& data, const MatShape& targetShape, int ax
     }
 }
 
-static void broadcastScaleAndZeropoint(Mat& scalesMat, Mat& zeropointsMat, const std::vector<float>& scales,
-                                       const std::vector<int>& zeropoints, const MatShape& targetShape, int axis)
+static void block_repeat(InputArray src, int axis, int repetitions, OutputArray dst)
 {
-    // broad cast the scales and zeropoint to the input shape.
-    MatShape subTargetShape(targetShape.size(), 1);
-    subTargetShape[axis] = scales.size();
+    CV_Assert(src.getObj() != dst.getObj());
+    CV_Check(axis, axis >= 0 && axis < src.dims(), "Axis out of range");
+    CV_CheckGT(repetitions, 1, "More than one repetition expected");
+
+    Mat src_mat = src.getMat();
+    Mat dst_mat;
+
+    MatShape sshape = shape(src_mat);
+    MatShape dshape = sshape;
+
+    size_t dtype_bytes = src_mat.elemSize();
+    int chunk_size = dtype_bytes;
+    int num_chunks = 1;
+
+    dshape[axis] *= repetitions;
+
+    for (int i = axis+1; i < src.dims(); ++i)
+        chunk_size*=sshape[i];
 
-    zeropointsMat.create(subTargetShape.size(), subTargetShape.data(), CV_32FC1);
-    scalesMat.create(subTargetShape.size(), subTargetShape.data(), CV_32FC1);
+    for (int i = 0; i <= axis; ++i)
+        num_chunks*=sshape[i];
 
-    const int len = scales.size();
-    // Deep copy the scales and zeropoint data and prevent the original data from being changed.
+    dst.create(src.dims(), dshape.data(), src.type());
+    dst_mat = dst.getMat();
+
+
+    for (int i = 0; i < repetitions; ++i) {
+        size_t src_offset = 0;
+        size_t dst_offset = i * chunk_size;
+
+        for (int j = 0; j < num_chunks; ++j) {
+            memcpy(dst_mat.data + dst_offset, src_mat.data + src_offset, chunk_size);
+            src_offset += chunk_size;
+            dst_offset += chunk_size * repetitions;
+        }
+    }
+}
+
+template <typename T>
+static void copyVecToMat(Mat& mat, const std::vector<T>& data){
+    float * matPtr = mat.ptr<float>(0);
+    const int len = data.size();
 
-    float * scalePtr = scalesMat.ptr<float>(0);
     for (int i = 0; i < len; i++)
-        scalePtr[i] = scales[i];
+        matPtr[i] = (float) data[i];
+}
 
-    float * zpPtr = zeropointsMat.ptr<float>(0);
+template <typename T>
+static void copyMatToVec(const Mat& mat, std::vector<T>& data){
```

Comment:
There is cv::Mat::copyTo, if you really need a copy.

Reply:
Thank you for the suggestion. I can see from the documentation `cv::Mat::copyTo` requires the destination to be a `cv::Mat`, but in this case the target is a `std::vector`. Is there anything I'm missing?

Reply:
destination is OutputArray. You can pass vector there too.